Code/Editor/AzAssetBrowser/AzAssetBrowserRequestHandler.cpp

```diff
@@ -146,105 +265,104 @@ namespace AzAssetBrowserRequestHandlerPrivate
             }
             else
             {
+                // non-slices, regular entities:
+
                 AZ::Uuid componentTypeId = AZ::Uuid::CreateNull();
                 AZ::AssetTypeInfoBus::EventResult(componentTypeId, product->GetAssetType(), &AZ::AssetTypeInfo::GetComponentTypeId);
                 if (!componentTypeId.IsNull())
                 {
-                    assetIdAndComponentTypeId.push_back(AssetIdAndComponentTypeId(componentTypeId,product->GetAssetId()));
+                    AZStd::string entityName;
 
-                    componentsToAdd.push_back(componentTypeId);
-                }
-            }
-        }
-
-        if (assetIdAndComponentTypeId.empty())
-        {
-            return;
-        }
-
-        ScopedUndoBatch undo("Create entities from asset");
-
-
-        AZStd::string entityName;
-
-        // If the entity is being created from an asset, name it after said asset.
-        // If there is more than one product, use the first to generate the name.
-        AZStd::string assetPath;
-        AZ::Data::AssetCatalogRequestBus::BroadcastResult(
-            assetPath, &AZ::Data::AssetCatalogRequests::GetAssetPathById, assetIdAndComponentTypeId[0].m_assetId);
-        if (!assetPath.empty())
-        {
-            AzFramework::StringFunc::Path::GetFileName(assetPath.c_str(), entityName);
-        }
+                    // If the entity is being created from an asset, name it after said asset.
+                    AZStd::string assetPath;
+                    AZ::Data::AssetCatalogRequestBus::BroadcastResult(
+                        assetPath, &AZ::Data::AssetCatalogRequests::GetAssetPathById, product->GetAssetId());
 
-        // If not sourced from an asset, generate a generic name.
-        if (entityName.empty())
-        {
-            entityName = AZStd::string::format("Entity%d", GetIEditor()->GetObjectManager()->GetObjectCount());
-        }
+                    if (!assetPath.empty())
+                    {
+                        AZStd::string assetPathFileName;
+                        AzFramework::StringFunc::Path::GetFileName(assetPath.c_str(), assetPathFileName);
+                        if (assetPathFileName.empty())
```

Comment:
FYI the AZ::IO::Path* classes support retrieving a non-const lvalue reference to the underlying string type via the [.Native()](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/Path/Path.h#L542-L545) function.
This allows populating the string that the path stores via EBus class

```c++
                    AZ::IO::Path assetPath;
                    AZ::Data::AssetCatalogRequestBus::BroadcastResult(
                        assetPath.Native(), &AZ::Data::AssetCatalogRequests::GetAssetPathById, product->GetAssetId());

                    if (!assetPath.empty())
                    {
                        AZ::IO::Path assetPathFileName = assetPath.Stem();
                        if (assetPathFileName.empty())
```

Reply:
I'll update it.  Frankly I might actually rewrite this part anyway, its frustrating to me that only the **source** asset browser entry actually supports full path.   You'd think the product would also have a full path.

Reply:
Yes, it seems that the UpdateChildPaths function should be overwritten by the ProductAssetBrowserEntry to set the product path to @products@ + <relative path>

Reply:
Yes, just concerned about breaking stuff.  I notice that for sources, the "relative path" and "name" are the file path and name, but for products, the full path is not an API, and the "name" is actually something like "Sphere (Model)" instead of say, "sphere.atom_model", which indicates its being used somehow somewhere not as a file name.  I'll do some investigation

Reply:
I fixed it all up, see latest PR